daily_report/daily_report.py

```diff
@@ -240,12 +239,41 @@ def create_html_report(template_file, content_dict):
         report_html = report_html.replace(content, content_dict[content])
         
 
-    
     return report_html
  
- 
-report_html = create_html_report("template.html", content_dict)
-
-with open('todays_report.html', 'w') as f:
-    f.write(report_html)
- 
+# def write_daily_summary():
+
+
+def lambda_handler(event,context):
+    today = str(datetime.now().date())
+    total_rides = get_total_daily_rides()
+    total_rides_html = figure_to_base64(total_rides)
+    avg_power = f"{(round(get_avg_power(), 2))} W"
+    total_ride_time = f"{get_total_ride_time('both')} s"
+    ride_time_m = f"{get_total_ride_time('Male')} s"
+    ride_time_f = f"{get_total_ride_time('Female')} s"
+    avg_hr = f"{math.floor(get_avg_hr())} bpm"
+    avg_ride_dur = get_avg_ride_length_by_gender_and_age()
+    avg_ride_dur_html = figure_to_base64(avg_ride_dur)
+    avg_ride_pow = get_avg_power_by_gender_and_age()
+    avg_ride_pow_html = figure_to_base64(avg_ride_pow)
+    over_max_hr = pull_txt_file_from_s3()
+    max_power = f"{round(get_largest_power(), 2)} W"
+    longest_ride = f"{get_longest_ride()} s"
+    logo_html = encode_logo_as_base64()
+
+    content_dict = {
+        '{{ date }}':today,'{{ total_rides_graph }}':total_rides_html,
+        '{{ avg_power }}':avg_power,'{{ total_ride_time }}':total_ride_time,
+        '{{ ride_time_m }}': ride_time_m, '{{ ride_time_f }}': ride_time_f, '{{ avg_hr }}':avg_hr,
+        '{{ avg_dur_graph }}':avg_ride_dur_html,'{{ avg_pow_graph }}':avg_ride_pow_html,
+        '{{ over_max_hr }}':over_max_hr,'{{ max_power }}':max_power,
+        '{{ longest_ride }}':longest_ride, '{{ logo }}': logo_html}
+
+    report_html = create_html_report("template.html", content_dict)
+
+    with open(f'{today}.html', 'w') as f:
+        f.write(report_html)
+
+
+lambda_handler('x', 'x')
```

Comment:
nice type hinting and clear separation of concerns with smaller more specialised functions